xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java

```diff
@@ -132,28 +132,30 @@
     @Override
     public ResourceReference parse(String rawReference)
     {
+        StringBuilder content = new StringBuilder(rawReference);
+        ResourceReference parsedURILinks = parseURILinks(rawReference);
+        ResourceReference parsedInterWikiLinks = parseInterWikiLinks(content);
```

Comment:
It does not make sense to do this before knowing if you will need it in the first place.